src/main/java/com/amazon/opendistroforelasticsearch/security/rest/TenantInfoAction.java

```diff
@@ -137,7 +144,35 @@ public void accept(RestChannel channel) throws Exception {
             }
         };
     }
-    
+
+    private boolean isAuthorized() {
+        final User user = (User)threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
+
+        // check if the user is a kibanauser or super admin
+        boolean validateUserByName = (user != null) && (user.getName().equals(evaluator.kibanaServerUsername()) || adminDns.isAdmin(user));
```

Comment:
instead of checking multiple times "user != null", can you add just one line above:
```
if user == null:
  return false;
```